File: 코드리뷰/정집민/1일차/코드리뷰_밑장빼기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class main {
    static boolean[][] cardSet = new boolean[4][13];
    public static void main(String[] args)throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String cards = br.readLine();
        boolean flag = false;
        
        
        for (int i = 0; i < cards.length()/3; i++) {
            char shape = cards.charAt(i*3);
            int shapeNumber;
            if(shape=='P') {
                shapeNumber=0;
            }else if(shape=='K') {
                shapeNumber=1;
            }else if(shape=='H') {
                shapeNumber=2;
            }else {
                shapeNumber=3;
            }
            int number = Integer.valueOf(cards.substring(i*3+1, i*3+3));
            if(!cardSet[shapeNumber][number]) {
                cardSet[shapeNumber][number]=true;
            }else {
                flag = true;
                break;
            }
        }
        
        if(flag) {
            System.out.println("GRESKA");
        }else {
            for (int i = 0; i < cardSet.length; i++) {
                int count = 0;
                for (int j = 0; j < 13; j++) {
                    if(!cardSet[i][j]) {
                        count++;
                    }
                }
                System.out.print(count + " ");
            }
        }
        

    }

}

File: 코드리뷰/장문주(jmj2359)/1일차/코드리뷰_밑장빼기.py
"""
count를 통해 해당 원소가 2개 이상 존재하는지 확인한 코드네요.
지금은 입력 사이즈가 작아서 상관없지만, 나중에는 count(i)가 O(N)에 해당 개수를 찾아주기 때문에
시간초과가 날 수도 있습니다. 존재 여부를 배열에 체크하는 방법도 숙지해두시면 좋겠습니다!
"""

n = input()
c = 0
l = []
d = {"P":13,"K":13,"H":13,"T":13}
for i in range(len(n)//3):
    c =2+3*i
    l.append(n[c-2:c+1])
for i in l:
    if l.count(i)>=2:
        print("GRESKA")
        break
    else:
        for i in l:
            d[i[0]]-=1
        for i in d.values():
            print(i, end=" ")
        print("")
        break
